recipes/dd-agent.rb

```diff
@@ -135,5 +135,14 @@ def template_vars
   retry_delay 5
 end
 
+# only load system-probe recipe if an agent6 installation comes with it
+ruby_block 'include system-probe' do
+  block do
+    if ::File.exist?('/opt/datadog-agent/embedded/bin/system-probe') && node['datadog']['agent6'] && !is_windows
```

Comment:
I'm not a huge fan of the binary existence test and not sure that it's actually needed. I would say that it's not really future proof compared to the rest of the recipe.
I think we could just consider adding a note in the README (such as the one there is about the logs feature).

Reply:
I think the `include_recipe` has to be guarded with this, otherwise installing any earlier versions of datadog-agent would break `system-probe` cookbook, because the service `system-probe` doesn't exist. Do you have better suggestions?

Reply:
I said above based on the fact that dd-agent would still include system-probe recipe.

Reply:
OK, if we don't want the user to manually add the recipe in the runlist, that's most likely the best option. We will have to think about this constant if some day the binary is renamed or moved!